napari/_tests/test_viewer.py

```diff
@@ -12,33 +12,41 @@
     skip_local_popups,
     skip_on_win_ci,
 )
+from napari.settings import get_settings
 from napari.utils._tests.test_naming import eval_with_filename
 from napari.utils.action_manager import action_manager
 
 
-def _get_all_keybinding_methods(type_):
-    obj_methods = set(super(type_, type_).class_keymap.values())
-    obj_methods.update({v.__name__ for v in type_.class_keymap.values()})
-    obj_methods.update(
-        {
-            a.command.__name__
-            for a in action_manager._get_layer_actions(type_).values()
-        }
-    )
-    return obj_methods
+def _get_provider_actions(type_):
+    return {
+        action.command.__name__
+        for action in action_manager._get_provider_actions(type_).values()
+    }
+
+
+def _assert_shortcuts_exist_for_each_action(type_):
+    actions = _get_provider_actions(type_)
+    shortcuts = {
+        name.partition(':')[-1] for name in get_settings().shortcuts.shortcuts
+    }
+    shortcuts.update(func.__name__ for func in type_.class_keymap.values())
+    for action in actions:
+        assert (
+            action in shortcuts
+        ), f"Shortcut for action '{action}' is missing"
 
 
-viewer_methods = _get_all_keybinding_methods(Viewer)
-EXPECTED_NUMBER_OF_VIEWER_METHODS = 14
+viewer_actions = _get_provider_actions(Viewer)
 
 
-def test_len_methods_viewer(make_napari_viewer):
+def test_all_viewer_actions_are_accessible_via_shortcut(make_napari_viewer):
     """
-    Make sure we do find all the methods attached to a viewer via keybindings
+    Make sure we do find all the actions attached to a viewer via keybindings
     """
-    _ = make_napari_viewer()
-    viewer_methods = _get_all_keybinding_methods(Viewer)
-    assert len(viewer_methods) == EXPECTED_NUMBER_OF_VIEWER_METHODS
+    _ = (
+        make_napari_viewer()
+    )  # instantiate to make sure everything is initialized correctly
+    _assert_shortcuts_exist_for_each_action(Viewer)
 
 
 @pytest.mark.xfail
```

Comment:
is this xfail required?

Reply:
Not sure, I left this untouched from before. I think the this boils down to the same issue below where you need to first instantiate the object for actions to be correctly initialized. I think we can ust remove it?

Reply:
Actually, probably better to leave it, it shows that we have some issue with intitialization so we don't forget to solve it at some point. I'd say solving it is out of scope for this PR, though.